examples/guide/main.go

```diff
@@ -30,9 +30,9 @@ func main() {
 		// http://<test-client>:8080/?behavior=dance
 		behavior := r.FormValue("behavior")
 
-		// when client is called with no arguments,
-		// report back with a 200 when ready to run tests
-		if behavior == "" {
+		// when method is HEAD, report back with a 200 when ready to run tests
+		if r.Method == "HEAD" {
+			w.Header().Add("Content-Length", "0")
```

Comment:
I stand corrected. I was tricked by curl -X HEAD vs curl --head :) In the former, curl doesn't understand the request as a HEAD request, but as a generic http request. Go http server is smart enough to fill the content-length when required.